neural_compressor/torch/algorithms/weight_only/modules.py

```diff
@@ -47,6 +48,170 @@ def forward(self, X):
 
 
 class WeightOnlyLinear(torch.nn.Module):
+    def __init__(
+        self, in_features, out_features, dtype, bits, group_size, bias, scale_dtype, compression_dtype, device
+    ):
+        super().__init__()
+        self.dtype = dtype
+        if self.dtype != "int" and "int" in self.dtype:  # for nf4, fp4
+            bits = int(self.dtype.lstrip("int"))
+            self.dtype = "int"
+        self.bits = bits
+        self.device = device
+        self.in_features = in_features
+        self.out_features = out_features
+        self.group_size = group_size if group_size != -1 else in_features
+        self.float_type = scale_dtype
+        assert compression_dtype in [
+            torch.int8,
+            torch.int16,
+            torch.int32,
+            torch.int64,
+        ], "Only support torch.int8|16|32|64 as compressed dtype."
```

Comment:
```suggestion
        ], f"Only support torch.int8|16|32|64 as compressed dtype. but got {compression_dtype}"
```

Reply:
Fixed in https://github.com/intel/neural-compressor/pull/1877/commits/ab45bf219689801e319214358e5184ef8a2c745c